File: csharp/TetrisGameBase/util/Dimension.cs
namespace hu.klenium.tetris.util
{
    public struct Dimension
    {
        public readonly int width;
        public readonly int height;
        public Dimension(int width, int height)
        {
            this.width = width;
            this.height = height;
        }

        public static implicit operator Dimension((int w, int h) tuple)
        {
            return new Dimension(tuple.w, tuple.h);
        }
    }
}

File: haskell/src/Util/Dimension.hs
module Util.Dimension (
  Dimension(..),
  plusDim,
  toPoint
) where

import Util.Point

data Dimension = Dimension {width :: Int, height :: Int}
  deriving (Eq, Show)

plusDim :: Point -> Dimension -> Point
plusDim (Point x1 y1) (Dimension w h) = Point (x1 + w) (y1 + h)

toPoint :: Dimension -> Point
toPoint (Dimension w h) = Point w h